### PR TITLE
Upgrade Jackson dependencies to 2.17.2

### PR DESCRIPTION
**Pull request by alzimmermsft**
# Description

Upgrades Jackson dependency versions from `2.13.5` to `2.17.2`.

# All SDK Contribution checklist:
- [x] **The pull request does not introduce [breaking changes]**
- [x] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [x] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md).**

## [General Guidelines and Best Practices](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md#developer-guide)
- [x] Title of the pull request is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md#building-and-unit-testing)
- [x] Pull request includes test coverage for the included changes.

### CODE REVIEW

File: sdk/batch/microsoft-azure-batch/pom.xml [hunk 1]
```diff
@@ -71,7 +71,7 @@
     <dependency>
       <groupId>com.fasterxml.jackson.core</groupId>
       <artifactId>jackson-core</artifactId>
-      <version>2.13.5</version> <!-- {x-version-update;com.fasterxml.jackson.core:jackson-core;external_dependency} -->
+      <version>2.15.4</version> <!-- {x-version-update;com.fasterxml.jackson.core:jackson-core;external_dependency} -->
```
Comment (alzimmermsft) [inline sdk/batch/microsoft-azure-batch/pom.xml:74]: TODO: Add special version for `microsoft-azure-batch` as it supports Java 7 and Jackson Core requires Java 8 starting with version 2.14.0

File: sdk/mediaservices/microsoft-azure-media/pom.xml [hunk 2]
```diff
@@ -108,17 +108,17 @@
     <dependency>
   	  <groupId>com.fasterxml.jackson.core</groupId>
   	  <artifactId>jackson-databind</artifactId>
-     <version>2.13.5</version> <!-- {x-version-update;com.fasterxml.jackson.core:jackson-databind;external_dependency} -->
+     <version>2.15.4</version> <!-- {x-version-update;com.fasterxml.jackson.core:jackson-databind;external_dependency} -->
```
Comment (alzimmermsft) [inline sdk/mediaservices/microsoft-azure-media/pom.xml:111]: Same TODO as `microsoft-azure-batch`

**Review by srnagar — APPROVED** [review]

**Comment by alzimmermsft** [comment]
Overriding check enforcer as flaky tests are failing unrelated to this change.

**Pull request merged**